[CI] CcrRepositoryIT.testCcrRepositoryFetchesSnapshotShardSizeFromIndexShardStoreStats

**Build scan**:
[https://gradle-enterprise.elastic.co/s/ys4oqzz22q5j4](https://gradle-enterprise.elastic.co/s/ys4oqzz22q5j4)

**Repro line**:

> ./gradlew ':x-pack:plugin:ccr:internalClusterTest' --tests "org.elasticsearch.xpack.ccr.CcrRepositoryIT.testCcrRepositoryFetchesSnapshotShardSizeFromIndexShardStoreStats" -Dtests.seed=3E1F56A97DF16F81 -Dtests.security.manager=true -Dtests.locale=bg -Dtests.timezone=Etc/GMT-6 -Druntime.java=8

**Reproduces locally?**:
Haven't tried yet.

**Applicable branches**:
master and 7.x

**Failure history**:
[Failed 4 times last week](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(branch),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'test:%22testCcrRepositoryFetchesSnapshotShardSizeFromIndexShardStoreStats%22'),sort:!(time,desc)))

**Failure excerpt**:
```
java.lang.AssertionError: 
Expected: <9533L>
     but: was null
	at __randomizedtesting.SeedInfo.seed([3E1F56A97DF16F81:3800E1220BE5AB1C]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.ccr.CcrRepositoryIT.lambda$testCcrRepositoryFetchesSnapshotShardSizeFromIndexShardStoreStats$9(CcrRepositoryIT.java:564)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:999)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:972)
	at org.elasticsearch.xpack.ccr.CcrRepositoryIT.testCcrRepositoryFetchesSnapshotShardSizeFromIndexShardStoreStats(CcrRepositoryIT.java:560)
```

--- a/x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java
+++ b/x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java
@@ -473,13 +473,18 @@ public class CcrRepositoryIT extends CcrIntegTestCase {
                 TimeValue.ZERO.getStringRep())), XContentType.JSON));
 
         final int numDocs = scaledRandomIntBetween(0, 1_000);
-        final BulkRequestBuilder bulkRequest = leaderClient().prepareBulk(leaderIndex, "_doc");
-        for (int i = 0; i < numDocs; i++) {
-            bulkRequest.add(new IndexRequest(leaderIndex).id(Integer.toString(i)).source("field", i));
+        if (numDocs > 0) {
+            final BulkRequestBuilder bulkRequest = leaderClient().prepareBulk(leaderIndex, "_doc");
+            for (int i = 0; i < numDocs; i++) {
+                bulkRequest.add(new IndexRequest(leaderIndex).id(Integer.toString(i)).source("field", i));
+            }
+            assertThat(bulkRequest.get().hasFailures(), is(false));
         }
-        assertThat(bulkRequest.get().hasFailures(), is(false));
 
-        final ForceMergeResponse forceMergeResponse = leaderClient().admin().indices().prepareForceMerge(leaderIndex).setFlush(true).get();
+        final ForceMergeResponse forceMergeResponse = leaderClient().admin().indices().prepareForceMerge(leaderIndex)
+            .setMaxNumSegments(1)
+            .setFlush(true)
+            .get();
         assertThat(forceMergeResponse.getSuccessfulShards(), equalTo(numberOfShards));
         assertThat(forceMergeResponse.getFailedShards(), equalTo(0));
         ensureLeaderGreen(leaderIndex);
@@ -564,7 +569,7 @@ public class CcrRepositoryIT extends CcrIntegTestCase {
                     assertThat(snapshotShardSize,
                         equalTo(indexStats.getIndexShards().get(shardId).getPrimary().getStore().getSizeInBytes()));
                 }
-            });
+            }, 60L, TimeUnit.SECONDS);
 
             blockCcrRestore.countDown();
             ensureFollowerGreen(followerIndex);
@@ -678,7 +683,7 @@ public class CcrRepositoryIT extends CcrIntegTestCase {
                     final long randomSize = randomNonNegativeLong();
                     assertThat(snapshotShardSizeInfo.getShardSize(primary, randomSize), equalTo(randomSize));
                 }
-            });
+            }, 60L, TimeUnit.SECONDS);
         } finally {
             transportServices.forEach(MockTransportService::clearAllRules);
         }

